nautobot/core/utils/module_loading.py

```diff
@@ -0,0 +1,90 @@
+from contextlib import contextmanager
+import importlib
+from importlib.util import find_spec, module_from_spec
+import logging
+import os
+import pkgutil
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def _temporarily_add_to_sys_path(path):
+    """
+    Allow loading of modules and packages from within the provided directory by temporarily modifying `sys.path`.
+
+    On exit, it restores the original `sys.path` value.
+    """
+    old_sys_path = sys.path.copy()
+    sys.path.insert(0, path)
+    try:
+        yield
+    finally:
+        sys.path = old_sys_path
+
+
+def import_modules_privately(path, module_path=None, should_raise=False):
+    """
+    Import modules from the filesystem without adding the path permanently to `sys.path`.
+
+    This is used for importing Jobs from `JOBS_ROOT` and `GIT_ROOT` in such a way that they remain relatively
+    self-contained and can be easily discarded and reloaded on the fly.
+
+    If you find yourself writing new code that uses this method, please pause and reconsider your life choices.
+
+    Args:
+        path (str): Directory path possibly containing Python modules or packages to load.
+        module_path (list): If set to a non-empty list, only modules matching the given chain of modules will be loaded.
+            For example, `["my_git_repo", "jobs"]`.
+        should_raise (bool): Whether any exception raised in importing modules should be re-raised to the caller.
```

Comment:
Same question as earlier, is this more clear if it's called `ignore_import_errors` instead of `should_raise`?